File: swift/0136-single-number.swift
//BitManipulation Solution O(n), SC(1)
class Solution {
    func singleNumber(_ nums: [Int]) -> Int {
      var ans = 0
      for i in nums {
          ans ^= i
      }
      return ans
    }
}

//HashMapSolution O(n), SC(n)
class Solution {
    func singleNumber(_ nums: [Int]) -> Int {
        var hm: [Int: Int]  = [:]
        for i in nums {
            hm[i,default: 0] += 1
        }
        for i in nums {
            if hm[i] == 1 { return i }
        }
        return -1
    }
}

File: dart/0136-single-number.dart
class Solution {
  int singleNumber(List<int> nums) {
    int? val = null;
    nums.sort();
    int i = 0;
    while (i < nums.length - 2) {
      if (!(nums[i] != nums[i + 2] && nums[i] == nums[i + 1])) {
        val = nums[i];
        break;
      } else {
        i = i + 2;
      }
    }
    if (val == null) {
      val = nums[nums.length - 1];
    }
    return val;
  }
}
